python/pyiceberg/cli/console.py

```diff
@@ -36,6 +37,23 @@
 SUPPORTED_CATALOGS: Dict[str, Type[Catalog]] = {"thrift": HiveCatalog, "http": RestCatalog}
 
 
+def catch_exception(func=None):
+    if not func:
+        return partial(catch_exception)
```

Comment:
I don't quite understand what's happening with this decorator. The wrapper part makes sense to me.

Reply:
I've just cleaned this up. This was a WIP and is required if you want to pass additional arguments along the decorator. This is easy by creating a partial that already binds the arguments. But this is actually not required. I've resolved the conflicts and simplified this code a bit.